src/fides/lib/oauth/roles.py

```diff
@@ -0,0 +1,100 @@
+from typing import Dict, List, Optional
+
+from fides.api.ops.api.v1.scope_registry import (
+    CLI_OBJECTS_READ,
+    CLIENT_READ,
+    CONFIG_READ,
+    CONNECTION_READ,
+    CONNECTION_TYPE_READ,
+    CONSENT_READ,
+    DATAMAP_READ,
+    DATASET_READ,
+    MESSAGING_READ,
+    POLICY_READ,
+    PRIVACY_REQUEST_CALLBACK_RESUME,
+    PRIVACY_REQUEST_CREATE,
+    PRIVACY_REQUEST_DELETE,
+    PRIVACY_REQUEST_NOTIFICATIONS_CREATE_OR_UPDATE,
+    PRIVACY_REQUEST_NOTIFICATIONS_READ,
+    PRIVACY_REQUEST_READ,
+    PRIVACY_REQUEST_REVIEW,
+    PRIVACY_REQUEST_TRANSFER,
+    PRIVACY_REQUEST_UPLOAD_DATA,
+    PRIVACY_REQUEST_VIEW_DATA,
+    RULE_READ,
+    SAAS_CONFIG_READ,
+    SCOPE_READ,
+    SCOPE_REGISTRY,
+    STORAGE_READ,
+    USER_READ,
+    WEBHOOK_READ,
+)
+
+PRIVACY_REQUEST_MANAGER = "privacy_request_manager"
+VIEWER = "viewer"
+VIEWER_AND_PRIVACY_REQUEST_MANAGER = "viewer_and_privacy_request_manager"
+ADMIN = "admin"
+
+
+ROLE_REGISTRY = {
+    PRIVACY_REQUEST_MANAGER,
+    VIEWER,
+    VIEWER_AND_PRIVACY_REQUEST_MANAGER,
+    ADMIN,
+}
+
+
+privacy_request_manager_scopes = [
+    PRIVACY_REQUEST_CREATE,
+    PRIVACY_REQUEST_REVIEW,
+    PRIVACY_REQUEST_READ,
+    PRIVACY_REQUEST_DELETE,
+    PRIVACY_REQUEST_CALLBACK_RESUME,
+    PRIVACY_REQUEST_NOTIFICATIONS_CREATE_OR_UPDATE,
+    PRIVACY_REQUEST_NOTIFICATIONS_READ,
+    PRIVACY_REQUEST_TRANSFER,
+    PRIVACY_REQUEST_UPLOAD_DATA,
+    PRIVACY_REQUEST_VIEW_DATA,
+]
```

Comment:
Which roles go with which scopes are still subject to change.  For example, I've given the Privacy Request Manager all the privacy request scopes, but there are likely requests the UI makes to load various parts of the UI that this role requires.

For example, there's a set of endpoints that aren't really related to resources created by the customer, but more system settings, like a list of all our connection types that might be loaded on various pages that make a tricky UI experience for the privacy request manager.

Reply:
I think this is a great starting point